gatsby-config.js

```diff
@@ -102,26 +119,48 @@ module.exports = {
               query: { site, storageRootConfig, allContentfulCampaign, allContentfulOneshot },
             }) => {
               const storageRoot = storageRootConfig.config.storageRoot
-              const formatEpisode = (title, episode) => {
+              const formatEpisode = (title, image, episode) => {
+                // Itunes only wants images larger than 1400x1400
+                const useImage = image.file.url && (image.file.details.image.width >= 1400) && (image.file.details.image.height >= 1400)
                 const url = makeStorageUrl(storageRoot, episode.filename)
                 const filename = (new URL(url)).pathname.replace(/^\//, '')
+                const episodeNr = filename.split('-', 2)[0]
+                const oldStyleGuid = 'https://www.svartvikenrp.se/?name=' + filename
+                var chosenGuid = filename
+                if (episode.pubDate && (Date.parse(episode.pubDate) < Date.parse("2021-09-15"))) {
+                  chosenGuid = oldStyleGuid // Use old style of guid for anything published before the website launch.
+                }
+                var totalseconds = 0
+                if (episode.duration) {
+                  const parts = episode.duration.split(":")
+                  if (parts.length === 2) {
+                    const minutes = parseInt(parts[0])
+                    const seconds = parseInt(parts[1])
+                    totalseconds = (isNaN(minutes) ? 0 : minutes * 60) + (isNaN(seconds) ? 0 : seconds)
+                  } else if (parts.length === 3) {
+                    const hours = parseInt(parts[0])
+                    const minutes = parseInt(parts[1])
+                    const seconds = parseInt(parts[2])
+                    totalseconds = (isNaN(hours) ? 0 : hours * 60 * 60) + (isNaN(minutes) ? 0 : minutes * 60) + (isNaN(seconds) ? 0 : seconds)
```

Comment:
I sann review anda tänker jag åkalla "rule of three". ;)

```javascript
const withDefault = default => value => isNan(value) ? default : value;
// usage
const withDefault0 = withDefault(0);
const minutes = withDefault0(parseInt(parts[0]);
```

Jag tycker detta är bra, vill mest bara visa att jag läst dina PR! :p